src/app/modules/shared/services/template.service.ts

```diff
@@ -86,11 +102,26 @@ export class TemplateService {
     return this.dataService.post(reqParam);
 
     }
-    getSurveySolutions(): Observable<any> {
-      const reqParam = {
-        url: 'survey/getSolutions'
-      };
-      return this.dataService.post(reqParam);
-    }
+
+  getSurveySolutions(resourceType: string): Observable<any> {
```

Comment:
getSurveySolutions and downloadSurveySolutions functions can be extracted into a single function send unique part of the url through the function argument